frontend/src/pages/budgetLines/PreviewTable.jsx

```diff
@@ -0,0 +1,163 @@
+import { useState, Fragment } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import Tag from "../../components/UI/Tag/Tag";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faChevronDown, faChevronUp, faPen, faTrash } from "@fortawesome/free-solid-svg-icons";
+import { faClock, faClone } from "@fortawesome/free-regular-svg-icons";
+import { deleteBudgetLineAdded, editBudgetLineAdded, duplicateBudgetLineAdded } from "./createBudgetLineSlice";
+import "./PreviewTable.scss";
+
+export const PreviewTable = ({ budgetLines }) => {
+    const dispatch = useDispatch();
+    const budgetLinesAdded = useSelector((state) => state.createBudgetLine.budget_lines_added);
+    const loggedInUser = useSelector((state) => state.auth.activeUser.full_name);
+
+    const TableRow = ({ bl }) => {
+        const [isExpanded, setIsExpanded] = useState(false);
+        const [isRowActive, setIsRowActive] = useState(false);
+        // create function to format date like this 9/30/2023 || MM/DD/YYYY
+        const formatDate = (date) => {
+            return `${date.getMonth() + 1}/${date.getDate()}/${date.getFullYear()}`;
+        };
+        let today = new Date();
+        const formatted_today = formatDate(today);
+        let date_needed = new Date(bl?.date_needed);
+        const formatted_date_needed = formatDate(date_needed);
+        // FY will automate based on the Need by Date. Anything after September 30th rolls over into the next FY.
+        let month = date_needed.getMonth();
+        let year = date_needed.getFullYear();
+        let fiscalYear = month > 8 ? year + 1 : year;
+        let feeTotal = bl?.amount * (bl?.psc_fee_amount / 10);
+        let total = bl?.amount + feeTotal;
+        let status = bl?.status.charAt(0).toUpperCase() + bl?.status.slice(1).toLowerCase();
+        // Format the amounts like this $500,000.00 || $1,000,000.00 to allow for commas
+        let formattedAmount = `$${bl?.amount.toFixed(2).replace(/\d(?=(\d{3})+\.)/g, "$&,")}`;
```

Comment:
I could see making a util function for the formatting, since we reuse it a lot. I thought we already had a currancyFormatter function, format but only seeing our use of `CurrencyFormat` from react-currency-format; Just want to ensure we're consistent.

Reply:
Was able to integrate `CurrencyFormat` in the table and amount input.